### PR TITLE
fix(Upload): gif thumb url base64 type should be corrected

### LINKED ISSUE #44044: 使用 Upload 组件上传 GIF 后打开只会播放一次
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/zhao-pian-qiang-antd-4-24-13-forked-4z8jyg?file=/demo.tsx)

### Steps to reproduce

https://media.tenor.com/qMZe3e1Er7EAAAAC/example.gif
本地下载该 gif 后点击 Upload，成功后点击 Preview 查看结果

### What is expected?

从本地上传的 GIF 经过 Upload 组件后应仍然保持循环播放的格式

### What is actually happening?

本地 GIF 支持循环播放，通过 Upload 组件上传后，返回的结果打开只会播放一次

| Environment | Info |
| --- | --- |
| antd | undefined |
| React | 18.2 |
| System | macOS |
| Browser | Chrome |

---
支持用户上传 GIF 图片，前端展示 GIF 图片（保留循环播放的效果）

### PR DESCRIPTION
**Pull request by linxianxi**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

fix https://github.com/ant-design/ant-design/issues/44044

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Fix the bug that Upload gif thumbnail will not play      |
| 🇨🇳 Chinese |  修复 Upload gif 缩略图不会动的问题        |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 1070873</samp>

This pull request adds support for previewing gif files in the `Upload` component. It modifies the `previewImage` function in `components/upload/utils.ts` and adds a test case in `components/upload/__tests__/uploadlist.test.tsx`.

### <samp>🤖 Generated by Copilot at 1070873</samp>

*  Add a new branch to handle gif files in `previewImage` function ([link](https://github.com/ant-design/ant-design/pull/44083/files?diff=unified&w=0#diff-9db1f0b51764a51b4b2f89c1faf12a7275921656aabbde4b55c88a7129940926R121-R126)) in `components/upload/utils.ts`
*  Add a test case to verify that gif files are converted to data URLs with image/gif MIME type in `uploadlist.test.tsx` ([link](https://github.com/ant-design/ant-design/pull/44083/files?diff=unified&w=0#diff-be2aa30e7597e9d996e3b0573a8f2f34568cbb899f203b3d270c95105132b390R924-R948))

### CODE REVIEW

File: components/upload/utils.ts [hunk 1]
```diff
@@ -118,6 +118,12 @@ export function previewImage(file: File | Blob): Promise<string> {
         if (reader.result) img.src = reader.result as string;
       });
       reader.readAsDataURL(file);
+    } else if (file.type.startsWith('image/gif')) {
```
[the remark this answers is not in the document]
Comment (linxianxi) [reply components/upload/utils.ts:121]: 这么快！其它走上面了，可以看下这个 pr 我新补充的描述

File: components/upload/utils.ts [hunk 2]
```diff
@@ -118,6 +118,12 @@ export function previewImage(file: File | Blob): Promise<string> {
...
+      const reader = new FileReader();
+      reader.addEventListener('load', () => {
+        if (reader.result) resolve(reader.result as string);
+      });
+      reader.readAsDataURL(file);
     } else {
       img.src = window.URL.createObjectURL(file);
```
Comment (tron-billy) [inline components/upload/utils.ts:128]: 辛苦了！恰好看到这行代码，window.URL.createObjectURL 会每次创建一个新的对象停留在内存里，应用关闭会自动清内存，但可能还是需要考虑下在 Upload 某个操作后调用 revokeObjectURL 主动释放内存。

Comment (linxianxi) [reply components/upload/utils.ts:128]: 确实，细

File: components/upload/utils.ts [hunk 3]
```diff
@@ -118,6 +118,12 @@ export function previewImage(file: File | Blob): Promise<string> {
         if (reader.result) img.src = reader.result as string;
       });
       reader.readAsDataURL(file);
+    } else if (file.type.startsWith('image/gif')) {
+      const reader = new FileReader();
+      reader.addEventListener('load', () => {
```
Comment (MadCcc) [inline components/upload/utils.ts:123]: 这个事件需要销毁么

Comment (linxianxi) [reply components/upload/utils.ts:123]: 好问题，我都改成 reader.onload = 把

**Review by afc163 — APPROVED** [review]

**Pull request merged**